### PR TITLE
feat: Add GMap option for error response callback

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/map/src/GMap.ts [hunk 1]
```diff
@@ -830,8 +844,11 @@ export interface GMap {
     showStreetViewMarker(_: boolean): this;
     outdoorStreetViewOnly(): boolean;
     outdoorStreetViewOnly(_: boolean): this;
+    useStatusErrorCallback(): boolean;
+    useStatusErrorCallback(_: boolean): this;
 }
 
+GMap.prototype.publish("useStatusErrorCallback", false, "boolean", "If true, api responses without status 200 will be passed as arguments to 'statusErrorCallback'");
```
Comment (GordonSmith) [inline packages/map/src/GMap.ts:851]: No need for "useStatusErrorCallback".

File: packages/map/src/GMap.ts [hunk 2]
```diff
@@ -770,6 +781,9 @@ export class GMap extends HTMLWidget {
                 JSON.stringify(this._userShapes.save()));
         }
     }
+    statusErrorCallback(response) {
+
```
Comment (GordonSmith) [inline packages/map/src/GMap.ts:785]: By default this method should have the default code:  `console.error("Street View data not found for this location.");`

File: packages/map/src/GMap.ts [hunk 3]
```diff
@@ -576,7 +583,11 @@ export class GMap extends HTMLWidget {
                     }
                 });
             } else {
-                console.error("Street View data not found for this location.");
+                if(context.useStatusErrorCallback()){
```
Comment (GordonSmith) [inline packages/map/src/GMap.ts:586]: No need for `useStatusErrorCallback` just call the default implementation (and move the message in there).

File: packages/map/src/GMap.ts [hunk 4]
```diff
@@ -770,6 +781,9 @@ export class GMap extends HTMLWidget {
                 JSON.stringify(this._userShapes.save()));
         }
     }
+    statusErrorCallback(response) {
```
Comment (GordonSmith) [inline packages/map/src/GMap.ts:784]: Rename to "statusError" and it should follow the "click" pattern.

File: demos/gallery/samples/geospatial/Google Maps/StreetView3.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    .useComputedHeading(true)
+    .streetViewControl(true)
+    .streetView(true)
+    .showStreetViewMarker(false)
+    .outdoorStreetViewOnly(true)
+    ;
+
+map.statusErrorCallback = (response) => {
+    console.log(response);
+};
```
Comment (GordonSmith) [inline demos/gallery/samples/geospatial/Google Maps/StreetView3.js:16]: Change to:
```
map.on("statusError", response=>{
  ...
});
```

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by jbrundage** [comment]
@GordonSmith this is ready

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**